app/cdap/components/WranglerV2/RecipeStepsTable/index.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { IconButton, Typography } from '@material-ui/core';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+import styled from 'styled-components';
+
+import DataTable, { DataTableContainer } from 'components/WranglerV2/DataTable';
+
+export const RecipeStepCellWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+export const RecipeStepsTableContainer = styled(DataTableContainer)`
+  &&& {
+    width: 460px;
+  }
+  .MuiTableCell-root:first-child {
+    width: 64px;
+  }
+  .MuiTableCell-root:last-child {
+    width: 396px;
+  }
+  .MuiTableCell-root:has(.MuiIconButton-root) {
+    vertical-align: text-top;
+  }
+  .MuiTableBody-root {
+    .MuiIconButton-root {
+      display: none;
+      padding: 0;
+      margin-top: 4px;
+      margin-right: 9px;
+      align-self: flex-start;
+    }
+    .MuiTableRow-root:hover {
+      .MuiIconButton-root {
+        display: block;
+      }
+
+      &:has(.MuiIconButton-root) {
+        .cell-content-div {
+          width: 80%;
+        }
+      }
+    }
+  }
+`;
+
+export const getTableBodyCell = ({ value }) => () => (
+  <Typography component="span" variant="body2">
+    {value}
+  </Typography>
+);
+
+export const getTableHeaderCell = (label: string) => () => (
+  <Typography component="span" variant="body1">
+    {label}
+  </Typography>
+);
+
+export default function RecipeStepsTable({ recipeSteps }) {
+  const [rows, setRows] = useState([]);
+
+  useEffect(() => {
+    const updatedSteps = recipeSteps.map((recipeStep: string, index: number) => ({
+      serialNumber: String(index),
+      recipeStep,
+    }));
+
+    setRows(updatedSteps);
+  }, []);
```

Comment:
shouldn't recipeSteps be a dependency of this effect (what happens when the prop `recipeSteps` changes)?
Also, why do we need this effect and the state rows in the first place ?

Couldn't it be written simply as 

```js
const rows = recipeSteps.map((recipeStep: string, index: number) => ({
  serialNumber: String(index),
  recipeStep,
}));
```

Reply:
updated the code
removed unnecessary useState and useEffect hooks from the component
assigned the map return value directly to `const rows`